src/platform/silabs/wifi/icd/BUILD.gn

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2025 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# TODO: Once the WifiInterface refactor is done and becomes a class, we will inject an interface instead of having them in same source_set
+
+import("//build_overrides/chip.gni")
+
+source_set("power-save-interface") {
+  sources = [ "${chip_root}/src/platform/silabs/wifi/icd/PowerSaveInterface.h" ]
+
+  public_deps = [ "${chip_root}/src/lib/core" ]
+}
+
+source_set("wifi-sleep-manager") {
+  sources = [
+    "${chip_root}/src/platform/silabs/wifi/icd/WifiSleepManager.cpp",
+    "${chip_root}/src/platform/silabs/wifi/icd/WifiSleepManager.h",
+
+    # TODO: Remove duplicated source files - This is a temporary workaround until the efr32_sdk.gni assert is refactored
+    #       This source is necessary because we can't use anything in the src/platform/silabs/wifi/BUILD.gn due to the efr32_sdk.gni import
+    #       The import causes an assert when building unit-tests due the silabs_board not being defined
```

Comment:
we can define an arbitrary board or set the string to unit-test and handle it this way

Reply:
We could in the unit test build.gn as a temporary solution - I'll test it out and see.

Reply:
done